src/components/screens/HistoryScreen/TransactionDetailsBottomSheetCustomContent.tsx

```diff
@@ -0,0 +1,170 @@
+import BigNumber from "bignumber.js";
+import {
+  calculateSwapRate,
+  renderActionIcon,
+  renderIconComponent,
+} from "components/screens/HistoryScreen/helpers";
+import { CreateAccountTransactionDetailsContent } from "components/screens/HistoryScreen/mappers/createAccount";
+import { PaymentTransactionDetailsContent } from "components/screens/HistoryScreen/mappers/payment";
+import { SorobanTransferTransactionDetailsContent } from "components/screens/HistoryScreen/mappers/soroban";
+import { SwapTransactionDetailsContent } from "components/screens/HistoryScreen/mappers/swap";
+import {
+  TransactionDetails,
+  TransactionStatus,
+  TransactionType,
+} from "components/screens/HistoryScreen/types";
+import { Button } from "components/sds/Button";
+import Icon from "components/sds/Icon";
+import { Text } from "components/sds/Typography";
+import { formatDate } from "helpers/date";
+import { stroopToXlm } from "helpers/formatAmount";
+import useAppTranslation from "hooks/useAppTranslation";
+import useColors from "hooks/useColors";
+import React from "react";
+import { Linking, View } from "react-native";
+
+interface TransactionDetailsBottomSheetCustomContentProps {
+  transactionDetails: TransactionDetails | null;
+}
+
+/**
+ * Renders the transaction details in the bottom sheet
+ */
+export const TransactionDetailsBottomSheetCustomContent: React.FC<
+  TransactionDetailsBottomSheetCustomContentProps
+> = ({ transactionDetails }) => {
+  const { themeColors } = useColors();
+  const { t } = useAppTranslation();
+
+  if (!transactionDetails) {
+    return null;
+  }
+
+  const fee = stroopToXlm(transactionDetails.fee).toString();
+  const formattedDate = formatDate({
+    date: transactionDetails?.operation.created_at ?? "",
+    includeTime: true,
+  });
+  const isSuccess = transactionDetails.status === TransactionStatus.SUCCESS;
+  const swapRate =
+    transactionDetails.transactionType === TransactionType.SWAP
+      ? calculateSwapRate(
+          Number(transactionDetails.swapDetails?.sourceAmount ?? 0),
+          Number(transactionDetails.swapDetails?.destinationAmount ?? 0),
+        )
+      : 0;
+  const formattedSwapRate = new BigNumber(swapRate).toFixed(2, 1);
+  const swapRateText = `1 ${transactionDetails.swapDetails?.sourceAssetCode} ≈ ${formattedSwapRate} ${transactionDetails.swapDetails?.destinationAssetCode}`;
```

Comment:
I think we should always try using `formatAssetAmount` or `formatFiatAmount` helpers whenever we're presenting amounts to users for consistency and also so we could easily localize it in the future. The way it's implemented now it'll always display dot "." as the decimal separator whereas it changes based on the language/locale. Wdyt?

Reply:
well at least for the right side of the Rate label where we will have decimal places, for the left side (where it's always "1 ABC") we could probably skip the formatter